eclass/nodejs.eclass

```diff
@@ -0,0 +1,305 @@
+# Copyright 2019-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: nodejs.eclass
+# @MAINTAINER:
+# Fco. Javier Félix <web@inode64.com>
+# @AUTHOR:
+# Fco. Javier Félix <web@inode64.com>
+# @SUPPORTED_EAPIS: 7 8
+# @BLURB: An eclass for build NodeJS projects
+# @DESCRIPTION:
+# An eclass providing functions to build NodeJS projects
+#
+# Changelog:
+#   Initial version from:
+#       https://github.com/gentoo/gentoo/pull/930/files
+#       https://github.com/samuelbernardo/ssnb-overlay/blob/master/eclass/npm.eclass
+#       https://github.com/gentoo-mirror/lanodanOverlay/blob/master/eclass/nodejs.eclass
+#       https://github.com/Tatsh/tatsh-overlay/blob/master/eclass/yarn.eclass
+
+#
+# Build package for node_modules:
+#   npm:
+#   npm install --audit false --color false --foreground-scripts --progress false --verbose --ignore-scripts
+#
+#   yarn:
+#   yarn install --color false --foreground-scripts --progress false --verbose --ignore-scripts
+#
+#   Create archive in tar:
+#   tar --create --auto-compress --file foo-1-node_modules.tar.xz foo-1/node_modules/
+
+case ${EAPI} in
+7 | 8) ;;
+*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
+esac
+
+if [[ -z ${_NODEJS_ECLASS} ]]; then
+_NODEJS_ECLASS=1
+
+# @ECLASS_VARIABLE: NODEJS_MANAGEMENT
+# @PRE_INHERIT
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Specify a package management
+# The default is set to "npm".
+: "${NODEJS_MANAGEMENT:=npm}"
```

Comment:
"MANAGEMENT"? little weird

Reply:
I think `NPM` would be an okay name here actually, thought it might be confusing.

Reply:
What name could be more suitable?

NODEJS_NPM, NODEJS_MANAGER, NODEJS_PACKAGE_MANAGER, NODEJS_PROJECT_MANAGER, ....

* npm - javascript package manager
* yarn is a package manager that doubles down as project manager